test/functional/test_framework/test_framework.py

```diff
@@ -157,6 +157,8 @@ def parse_args(self):
                             help="use bitcoin-cli instead of RPC for all commands")
         parser.add_argument("--perf", dest="perf", default=False, action="store_true",
                             help="profile running nodes with perf for the duration of the test")
+        parser.add_argument("--valgrind", dest="valgrind", default=False, action="store_true",
+                            help="run nodes under the valgrind memory error detector: expect at least a ~10x slowdown, valgrind 3.14 or later required")
```

Comment:
what do you think about updating the [developer docs](https://github.com/bitcoin/bitcoin/blob/33c103e2fe67e32e9f4e783fa243dd2bba6b1682/doc/developer-notes.md) or some other documentation (I'm new to bitcoin so I don't know where the right place is) with instructions on valgrind? 

Is there a CI step that runs the functional tests? Should we use this option there?

Reply:
Searching for valgrind doesn't show much documentation: https://github.com/bitcoin/bitcoin/search?q=valgrind&unscoped_q=valgrind